src/org/rascalmpl/library/lang/rascal/grammar/definition/Priorities.rsc

```diff
@@ -105,26 +105,27 @@ public DoNotNest doNotNest(Grammar g) {
                            , same(ss, rr), same(t, lr), same(ss, t)}
         ; 
         
-     // and we warn about recursive productions which have been left ambiguous:
-    allProds  = {p | /p:prod(_,_,_) := g.rules[s]};
-    ambiguous = {<p, q>  | p:prod(Symbol ss, [Symbol lr, *_], _) <- allProds, same(s, lr),
-                           q:prod(Symbol t, [*_, Symbol rr], _) <- allProds,
-                            same(t, rr), same(ss, t)};
-    ambiguous += {<p, q> | p:prod(Symbol ss, [*_, Symbol rr], _) <- allProds, same(s, rr), 
-                           q:prod(Symbol t, [Symbol lr,   *_], _) <- allProds,
-                           same(t, lr), same(ss, t), <q, p> notin ambiguous}
-              ;
+    // and we warn about recursive productions which have been left ambiguous:
+    // TODO: this analysis can be done statically in the type-checker
```

Comment:
Yeah, you're right! But this generated a lot of spurious output while testing, and breaks the UX for the progress bars because of it. In the meantime these dynamic tests have been taken over by the static Rascal checker so removing this was on the backlog already.